Plugins/Public/dronebays/UserCommands.cpp

```diff
@@ -0,0 +1,377 @@
+
+	// All of the required information is present! Build the timer struct and add it to the list
+	DroneBuildTimerWrapper wrapper;
+	wrapper.buildTimeRequired = clientDroneInfo[iClientID].droneBay.iDroneBuildTime;
+	wrapper.reqDrone = requestedDrone;
+	wrapper.startBuildTime = timeInMS();
+
+	buildTimerMap[iClientID] = wrapper;
+
+	// Set the buildstate, and alert the user
+	clientDroneInfo[iClientID].buildState = STATE_DRONE_BUILDING;
+	PrintUserCmdText(iClientID, L"Drone being prepared for deployment :: Launch in ETA %i seconds", clientDroneInfo[iClientID].droneBay.iDroneBuildTime);
+
+	// Save the carrier shipObj to the client struct
+	clientDroneInfo[iClientID].carrierShipobj = playerShip;
+	return true;
+}
+
+bool UserCommands::UserCmd_AttackTarget(uint iClientID, const wstring& wscCmd, const wstring& wscParam,
+	const wchar_t* usage)
+{
+	// Verify that the user is in space
+	uint iShipObj;
+	pub::Player::GetShip(iClientID, iShipObj);
+	if(!iShipObj)
+	{
+		PrintUserCmdText(iClientID, L"You must be in space to use this command");
+		return true;
+	}
+
+	// Verify that the user has a drone currently deployed
+	if(clientDroneInfo[iClientID].deployedInfo.deployedDroneObj == 0)
+	{
+		PrintUserCmdText(iClientID, L"You must have a drone deployed for this to work");
+		return true;
+	}
+
+	// Get the players current target
+	uint iTargetObj;
+	pub::SpaceObj::GetTarget(iShipObj, iTargetObj);
+
+	if(!iTargetObj)
+	{
+		PrintUserCmdText(iClientID, L"Please target the vessel which the drones should be directed to");
+		return true;
+	}
+
+	const uint droneObj = clientDroneInfo[iClientID].deployedInfo.deployedDroneObj;
+
+	// Set the old target to neutral reputation, and the hostile one hostile.
+	Utility::SetRepNeutral(droneObj, clientDroneInfo[iClientID].deployedInfo.lastShipObjTarget);
+	Utility::SetRepHostile(droneObj, iTargetObj);
+
+	// Set the target hostile to the drone as well only if it isn't another existing drone 
+	bool isTargetDrone = false;
+	for (const auto& currClient : clientDroneInfo)
+	{
+		if (iTargetObj == currClient.second.deployedInfo.deployedDroneObj)
+		{
+			isTargetDrone = true;
+			break;
+		}
+	}
+
+	if (!isTargetDrone)
+	{
+		Utility::SetRepHostile(iTargetObj, droneObj);
+	}
+
+	clientDroneInfo[iClientID].deployedInfo.lastShipObjTarget = iTargetObj;
+	PrintUserCmdText(iClientID, L"Drone targeting selected ship\n");
+
+	// If the last shipObj the drone was targeting is a player, log the event
+	if (clientDroneInfo[iClientID].deployedInfo.lastShipObjTarget != 0)
+	{
+		const wstring charname = reinterpret_cast<const wchar_t*>(Players.GetActiveCharacterName(iClientID));
+		const uint targetid = HkGetClientIDByShip(clientDroneInfo[iClientID].deployedInfo.lastShipObjTarget);
+		const wstring targetname = reinterpret_cast<const wchar_t*>(Players.GetActiveCharacterName(targetid));
+
+		// Only bother logging if we weren't engaging a NPC
+		if (!targetname.empty())
```

Comment:
I wonder if this should be checking the value of targetid instead but whatever

Reply:
I needed the target name for functions later. Thus it was used here because I already pulled the variable.

Reply:
targetname can be gotten inside the if block